src/poetry/core/packages/package.py

```diff
@@ -283,18 +280,11 @@ def python_versions(self, value: str) -> None:
 
         self._python_versions = value
         self._python_constraint = constraint
-        self._python_marker = parse_marker(
-            create_nested_marker("python_version", self._python_constraint)
-        )
 
     @property
     def python_constraint(self) -> VersionConstraint:
         return self._python_constraint
 
-    @property
-    def python_marker(self) -> BaseMarker:
-        return self._python_marker
-
```

Comment:
I think I am still going to deprecate it instead of removing it because we already have released a `poetry-plugin-export` version, which claims to be compatible with Poetry 2.0 - and really is so far if I have not missed anything - but would not if we removed this property.